fdbserver/Resolver.actor.cpp

```diff
@@ -361,20 +365,20 @@ ACTOR Future<Void> resolveBatch(Reference<Resolver> self, ResolveTransactionBatc
 
 		// If shardChanged at or before this commit version, the proxy may have computed
 		// the wrong set of groups. Then we need to broadcast to all groups below.
-		stateTransactionsPair.first = toCommit.isShardChanged();
+		stateTransactionsPair.first = toCommit && toCommit->isShardChanged();
 		bool shardChanged = self->recentStateTransactionsInfo.applyStateTxnsToBatchReply(
-		    &reply, firstUnseenVersion, req.version, toCommit.isShardChanged());
+		    &reply, firstUnseenVersion, req.version, toCommit && toCommit->isShardChanged());
```

Comment:
nit- rather than checking `toCommit`, consider simplifying by calling `self->recentStateTransactionsInfo.applyStateTxnsToBatchReply` with `null` if `SERVER_KNOBS->PROXY_USE_RESOLVER_PRIVATE_MUTATIONS` is off, and `toCommit->isShardChanged()` otherwise.

Reply:
calling with `null` is not different, I think.